### PR TITLE
CRUD/action_buttons.html.twig  object check fix

### PR DESCRIPTION
**Pull request by nnmer**
## Changelog

```markdown
### Fixed
 - object variable inside action_buttons template doesn’t submitted correctly to admin.getActionButtons if the object is a child
```
## Subject

If the `object` doesn't have parent (admin class doesn't have relation to parent admin class) then it works correctly, but if that is a child then `object` variable inside of CRUD/action_buttons.html.twig template doesn’t submitted correctly to admin.getActionButtons. It submitted as `null`  when the object really exist

### CODE REVIEW

File: Resources/views/CRUD/action_buttons.html.twig [hunk 1]
```diff
@@ -9,7 +9,7 @@ file that was distributed with this source code.
 
 #}
 {% spaceless %}
-    {% for item in admin.getActionButtons(action, object|default(null)) %}
+    {% for item in admin.getActionButtons(action, (object is defined and object is not null)?object:null ) %}
```
Comment (OskarStark) [inline Resources/views/CRUD/action_buttons.html.twig]: does this really change anything?

If the object is defined we already take it, if not we set null by default, IMO your code does exactly the same, correct me if I am wrong

Comment (nnmer) [inline Resources/views/CRUD/action_buttons.html.twig]: well, yes, it looks that code is the same, but the reality is if i have child object and building the action_buttons then with `object|default(null)` i always getting null in the `getActionButtons` function. It work correctly if the object is a parent or have no parent.

But `(object is defined and object is not null)?object:null ` make it work correctly for both parent and child objects.


now digging inside the twig. Note, my object's entity __toString returns empty string (this is correct, this is how i need that)
```
function _twig_default_filter($value, $default = '')
{
    if (twig_test_empty($value)) {
        return $default;
    }

    return $value;
}
...
function twig_test_empty($value)
{
    if ($value instanceof Countable) {
        return 0 == count($value);
    }

    if (is_object($value) && method_exists($value, '__toString')) {
        return '' === (string) $value;
    }

    return '' === $value || false === $value || null === $value || array() === $value;
}
```

the point here is when this happens`object|default(null)` the `twig_test_empty` returns `true`, and if that is `true` the `_twig_default_filter` returns `default` value which is `null`. That's why even i have correct object class i still receiving the `null` inside the `getActionButtons` which later return only button for the `list` action

Comment (OskarStark) [inline Resources/views/CRUD/action_buttons.html.twig]: sounds reasonable to me, thank you for the detailed explanation 👍

Comment (jlamur) [inline Resources/views/CRUD/action_buttons.html.twig]: Isn't `object is defined ? object : null` sufficient?

Comment (Soullivaneuh) [inline Resources/views/CRUD/action_buttons.html.twig]: * Add a Twig comment to quickly explain why we should not use `default`
* Add spaces on your ternary condition: https://twig.sensiolabs.org/doc/2.x/coding_standards.html

Comment (jlamur) [inline Resources/views/CRUD/action_buttons.html.twig]: Replace with `(object is defined ? object : null)`.

**Comment by nnmer** [comment]
not relevant, see #4768

**Pull request closed without merging**